src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -8,111 +8,69 @@
-                    throw CreateSslException("Failed to allocate SSL/TLS context");
-                }
-
-                libssl.SSL_CTX_ctrl(contextPtr, libssl.SSL_CTRL_OPTIONS, options, IntPtr.Zero);
-
-                libssl.SSL_CTX_set_quiet_shutdown(contextPtr, 1);
-
-                if (certHandle != null && certKeyHandle != null)
+                using (libssl.SafeSslContextHandle sslContext = new libssl.SafeSslContextHandle(method))
                 {
-                    SetSslCertificate(contextPtr, certHandle, certKeyHandle);
-                }
-
-                sslContext.sslPtr = libssl.SSL_new(contextPtr);
+                    if ((null == sslContext) || sslContext.IsInvalid)
+                    {
+                        throw CreateSslException("Failed to allocate SSL/TLS context");
+                    }
 
-                libssl.SSL_CTX_free(contextPtr);
+                    libssl.SSL_CTX_ctrl(sslContext, libssl.SSL_CTRL_OPTIONS, options, IntPtr.Zero);
 
-                if (IntPtr.Zero == sslContext.sslPtr)
-                {
-                    throw CreateSslException("Failed to create SSSL object from SSL context");
-                }
+                    libssl.SSL_CTX_set_quiet_shutdown(sslContext, 1);
 
-                IntPtr memMethod = libcrypto.BIO_s_mem();
+                    if (certHandle != null && certKeyHandle != null)
+                    {
+                        SetSslCertificate(sslContext, certHandle, certKeyHandle);
+                    }
 
-                if (IntPtr.Zero == memMethod)
-                {
-                    throw CreateSslException("Failed to return memory BIO method function");
+                    context = new SafeSslHandle(sslContext, isServer);
                 }
 
-                sslContext.readBioPtr = libssl.BIO_new(memMethod);
-                sslContext.writeBioPtr = libssl.BIO_new(memMethod);
-
-                if ((IntPtr.Zero == sslContext.readBioPtr) || (IntPtr.Zero == sslContext.writeBioPtr))
-                {
-                    FreeBio(sslContext);
-                    throw CreateSslException("Failed to retun new BIO for a given method type");
-                }
-
-                if (isServer)
-                {
-                    libssl.SSL_set_accept_state(sslContext.sslPtr);
-                }
-                else
+                if (context.IsInvalid)
                 {
-                    libssl.SSL_set_connect_state(sslContext.sslPtr);
-                }
-
-                libssl.SSL_set_bio(sslContext.sslPtr, sslContext.readBioPtr, sslContext.writeBioPtr);
+                    throw CreateSslException("Failed to create SSL object from SSL context");
+                }      
             }
             catch
             {
-                Disconnect(sslContext.sslPtr);
-                throw;
+                if (null != context)
+                {
+                    context.Dispose();
+                    context = null;
+                }
```

Comment:
This used to throw (continue) the exception, now it's eating it.  If eating it is intentional (which I'm sure will get lots of disagreement) that should have a comment explaining why.